src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_secrets.rs

```diff
@@ -153,21 +218,33 @@ pub fn generate_common_annotated_test_key(
         base64_encoded_signature.to_lowercase()
     };
 
-    let mut key = String::new();
+    let mut key: String;
 
     loop {
         let key_length_in_bytes = 66;
         let mut key_bytes = vec![0; key_length_in_bytes];
 
         if let Some(test_char) = test_char {
-            key = format!("{:86}==", test_char.to_string().repeat(86));
-        } else {
+            key = format!("{:85}Q==", test_char.to_string().repeat(85));
+        } 
+        else {
             let mut rng = rand::thread_rng();
-            rng.fill_bytes(&mut key_bytes);
+            rng.try_fill_bytes(&mut key_bytes).expect("Failed to generate random bytes.");
+
+            key  = base_62::encode(&key_bytes);
+
+            if key.len() < 86 {
+                return Err(format!("The key length is less than 86 characters: {}", key));
```

Comment:
>characters [](http://example.com/codeflow?start=67&length=10)

you need to make a call. either add a period at the end of all your exception messages or don't. :)

Between this line and line 246 below.

NIT

Reply:
Fair call. Will fix this. Won't leave it for later.